targets/TARGET_NUVOTON/TARGET_M2351/device/StdDriver/m2351_crypto.c

```diff
@@ -1765,6 +1755,1734 @@ int32_t  ECC_VerifySignature(CRPT_T *crpt, E_ECC_CURVE ecc_curve, char *message,
     return ret;
 }
 
+#if XOM_SUPPORT // To support XOM ready curve table
+
+int32_t CurveCpy(unsigned int *p32, E_ECC_CURVE id)
+{
+    int32_t i;
+
+    switch(id)
+    {
+        case CURVE_P_192:
+            p32[  0] = 0x00000000;
```

Comment:
Is this really a code-size win over just `memcpy`ing from the const `ECC_CURVE` table?

Even if it is, I suspect you could get an equivalent saving by just changing the `char Ea[144]` etc members to `const char *Ea`, which removes the need to store zero-padding.

Reply:
The above code is for XOM (execute-only ROM). On mbed-os, it is disabled (`XOM_SUPPORT` == 0).

> Beyond that, why store in hex rather than binary anyway?

The curve definition source is in hex form. No further conversion.

Reply:
> The above code is for XOM (execute-only ROM).

Okay, not relevant here, but a bit puzzled. Maybe there's something I'm missing.

If you're having problems with not being able to access `const` data in execute-only ROM, there's something wrong with your memory map and/or linker/compiler setup. Execute-only ROM shouldn't be breaking basic concepts like being able to access `const` data objects.

If really necessary in some environment, the build should be placing read-only data regions in RAM via the linker map.

Reply:
@kjbracey-arm Why `const` data not place in normal ROM? This is security consideration. The major purpose of XOM is non-readable. If XOM code references `const` data in another ROM, compromising the ROM means compromising the XOM indirectly.

Anyway, XOM is not used on mbed-os, so just ignore it.